### DISCUSSION TITLE
Adding an Extra Collider

### DISCUSSION THREAD

**Discussion by cemalgnlts** [excaliburjs/Excalibur#2731] [Q&A] [answered]
Hi,

My bomb object already has a collider so it stays on the floor, but when it explodes I want to add a circle collider so I can access the objects within the area it exploded.

There is information about changing it in the document, but I want to add one more.

*(I saw the `Trigger` class but wanted to know if it is added to the actor object.)*

**Comment by cemalgnlts** [comment]
I misunderstood the Trigger class a bit. The `Trigger` indicates that it has collided with another object. I need more control.

I tried adding an Actor instead:
```js
// Capture objects colliding with the circle.
const actor = new ex.Actor({
  pos: this.pos,
  radius: 45,
  collisionType: ex.CollisionType.Passive,
  collider: ex.Shape.Circle(50),
  color: ex.Color.Black,
});

actor.on("postcollision", (ev) => console.log(ev));

this.scene.add(actor);
```

But this way I can't listen to the collision event. Shouldn't `ex.CollisionType.Passive` help listen for collision event?

**Reply by eonarheim** [reply, marked as answer]
@cemalgnlts Good question, this is definitely the approach I would suggest for your bomb using another actor/entity with a circle collider. You could add an additional trigger/actor to the top level scene, or as a child on the bomb actor.

When the [collision type](https://excaliburjs.com/docs/collisiontypes/#collision-type-behavior-matrix) is `ex.CollisionType.Passive` the `postcollision` event doesn't fire because no resolution was performed.

You can use the `collisionstart` and `collisionend` events, here is an example of using this kind of approach to enable 1-way platforms. 
https://github.com/excaliburjs/Excalibur/discussions/2577#discussioncomment-4990445

There is more documentation about the specific events here https://excaliburjs.com/docs/collision-events/#collision-start-collisionstart

**Reply by cemalgnlts** [reply]
Thanks,
I'm glad to see this example, I just completed it like this:
```js
const colliderComp = new ex.ColliderComponent(ex.Shape.Circle(50));
const bodyComp = new ex.BodyComponent();

const explosion = new ex.Entity([
  colliderComp,
  bodyComp
], "explode");

bodyComp.collisionType = ex.CollisionType.Passive;
bodyComp.group = ex.CollisionGroup.collidesWith([
  ex.CollisionGroupManager.groupByName("player")
]);

colliderComp.events.on("collisionstart", (ev) => this.onExplodeInteract(ev));

this.addChild(explosion);
```
